tap_core/src/adapters/test/receipt_storage_adapter_mock.rs

```diff
@@ -82,12 +59,55 @@ impl ReceiptStorageAdapter for ReceiptStorageAdapterMock {
             .map(|(&id, rx_receipt)| (id, rx_receipt.clone()))
             .collect())
     }
-    fn retrieve_receipts_upto_timestamp(
+    pub fn retrieve_receipts_upto_timestamp(
         &self,
         timestamp_ns: u64,
-    ) -> Result<Vec<(u64, ReceivedReceipt)>, Self::AdapterError> {
+    ) -> Result<Vec<(u64, ReceivedReceipt)>, AdpaterErrorMock> {
         self.retrieve_receipts_in_timestamp_range(..=timestamp_ns)
     }
+    pub fn remove_receipt_by_id(&mut self, receipt_id: u64) -> Result<(), AdpaterErrorMock> {
+        let mut receipt_storage =
+            self.receipt_storage
+                .write()
+                .map_err(|e| AdpaterErrorMock::AdapterError {
+                    error: e.to_string(),
+                })?;
+        receipt_storage
+            .remove(&receipt_id)
+            .map(|_| ())
+            .ok_or(AdpaterErrorMock::AdapterError {
+                error: "No receipt found with ID".to_owned(),
+            })
+    }
+    pub fn remove_receipts_by_ids(&mut self, receipt_ids: &[u64]) -> Result<(), AdpaterErrorMock> {
+        for receipt_id in receipt_ids {
+            self.remove_receipt_by_id(*receipt_id)?;
+        }
+        Ok(())
+    }
+}
+
+use thiserror::Error;
+#[derive(Debug, Error)]
+pub enum AdpaterErrorMock {
```

Comment:
Typo

Reply:
Also do we want to merge it like this? Maybe can rework the error now as well, since it will be another breaking change when/if we rework this error

Reply:
Yeah I conserved the typo, wasn't sure if this PR was the best place to fix it

Reply:
OK let me fix it then

Reply:
ecacbdd

Reply:
thanks!